schema_generator/search.py

```diff
@@ -77,30 +78,44 @@ def _derived_options_transformation_generator(checkbox_question):
 }
 
 
-def get_transformations(framework_slug):
-    for question in _get_questions_by_type(framework_slug, TRANSFORMATION_GENERATORS.keys()):
+def get_transformations(framework_slug, doc_type):
+    for question in _get_questions_by_type(framework_slug, doc_type, TRANSFORMATION_GENERATORS.keys()):
         for transformer in TRANSFORMATION_GENERATORS[question.type](question):
             yield transformer
 
 
-def generate_search_mapping(framework_slug, file_handle, mapping_type, extra_meta={}):
-    with open(os.path.join(_base_dir, 'frameworks', framework_slug, 'search_mapping.json'), 'r') as h_template:
+def generate_search_mapping(framework_slug, doc_type, file_handle, mapping_type, extra_meta={}):
+    with open(os.path.join(
+        _base_dir,
+        "frameworks",
+        framework_slug,
+        "search_mappings",
+        "{}.json".format(doc_type),
+    ), 'r') as h_template:
         mapping_json = json.load(h_template, object_pairs_hook=OrderedDict)  # preserve template order for git history
 
-    mapping_json['mappings'][mapping_type]['_meta'].update(extra_meta)
-    mapping_json['mappings'][mapping_type]['_meta']['transformations'] = list(chain(
-        mapping_json['mappings'][mapping_type]['_meta'].pop("transformations", ()),
-        get_transformations(framework_slug),
+    original_meta = mapping_json['mappings'][mapping_type].get("_meta", {})
```

Comment:
Does this need to be a copy so it's not mutated by what happens below?

Reply:
ah (you're feeding these to me aren't you?) but "what's written below" is written in a functional style so we *know* (within about 99% probability) that it *doesn't* have any weird side effects like mutating any of its operands.

(the missing 1% is developers who are completely insane and sneak mutating code into it anyway but occurrences of that should stand out like a sore thumb)

Reply:
lol

Reply:
I see. Nice.